fix(VDialog/VMenu): enable pointer-events for disabled activator

jacekkarczmarczyk opened the pull request:
## Description
Don't render contents and remove `pointer-events: none` for disabled menu/dialog

## Motivation and Context
fixes #5472

## How Has This Been Tested?
visuall, updated unit

## Markup:
<!--- Paste markup for testing your change --->
<details>

```vue
<template>
  <v-app>
    <v-content>
      <v-checkbox label="Disabled" v-model="disabled"></v-checkbox>
      <v-menu :disabled="disabled">
        <v-btn slot="activator">Menu</v-btn>
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-menu :disabled="disabled" open-on-hover>
        <v-btn slot="activator">Menu (hover)</v-btn>
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-btn id="external-disabled">External activator</v-btn>
      <v-menu :activator="activators['external-disabled']" :disabled="disabled">
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-btn id="external-disabled-hover">External activator (hover)</v-btn>
      <v-menu open-on-hover :activator="activators['external-disabled-hover']" :disabled="disabled">
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-dialog :disabled="disabled">
        <v-btn slot="activator">Dialog</v-btn>
        <v-card><v-card-text>I'm dialog</v-card-text></v-card>
      </v-dialog>
      <v-bottom-sheet :disabled="disabled">
        <v-btn slot="activator">Bottom sheet</v-btn>
        <v-card><v-card-text>I'm bottom sheet</v-card-text></v-card>
      </v-bottom-sheet>
      <v-subheader>Active</v-subheader>
      <v-menu>
        <v-btn slot="activator">Menu</v-btn>
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-menu open-on-hover>
        <v-btn slot="activator">Menu (hover)</v-btn>
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-btn id="external-enabled">External activator</v-btn>
      <v-menu :activator="activators['external-enabled']">
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-btn id="external-enabled-hover">External activator (hover)</v-btn>
      <v-menu open-on-hover :activator="activators['external-enabled-hover']">
        <v-card><v-card-text>I'm menu</v-card-text></v-card>
      </v-menu>
      <v-dialog>
        <v-btn slot="activator">Dialog</v-btn>
        <v-card><v-card-text>I'm dialog</v-card-text></v-card>
      </v-dialog>
      <v-bottom-sheet>
        <v-btn slot="activator">Bottom sheet</v-btn>
        <v-card><v-card-text>I'm bottom sheet</v-card-text></v-card>
      </v-bottom-sheet>
    </v-content>
  </v-app>
</template>

<script>
export default {
  data: () => ({
    activators: {},
    disabled: true
  }),
  mounted () {
    ['external-disabled', 'external-disabled-hover', 'external-enabled', 'external-enabled-hover'].forEach(id => {
      this.$set(this.activators, id, document.getElementById(id))
    })
  }
}
</script>
```
</details>

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

KaelWD commented on packages/vuetify/src/components/VDialog/VDialog.js:
```diff
@@ -145,6 +145,77 @@ export default {
     },
+    genActivator () {
+      if (!this.$slots.activator) return null
+      if (this.disabled) return this.$slots.activator
```
This will cause different layout behaviour depending on if the dialog is disabled or not.

jacekkarczmarczyk commented:
Update: added tests and external activator support, updated Playground.vue. The button for menu with external activator and open-on-hover prop doesn't work though because of https://github.com/vuetifyjs/vuetify/issues/5860
